### PR TITLE
Sp 421 start new plan page

### PR DESCRIPTION
**Pull request by Coupey**

### CODE REVIEW

File: app/plans/get.controller.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  let activePlanPresent = false
+  plans.forEach(plan => {
+    if (!plan.completedDate) {
+      activePlanPresent = true
+    }
+  })
+  return activePlanPresent
+}
+
+function isEmptyObject(obj) {
```
Comment (jonhindle) [inline app/plans/get.controller.js]: Feels like a generic toolkit utility function. Should we extract to a utility file maybe?

File: app/plans/get.controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  return !Object.keys(obj).length
+}
+
+const sentencePlanSummary = async (req, res) => {
```
Comment (jonhindle) [inline app/plans/get.controller.js]: We only ever use id. Worth destructuring out `{ params: id }`

File: app/plans/get.controller.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+const controller = require('./get.controller')
```
Comment (jonhindle) [inline app/plans/get.controller.test.js]: destructure here? { sentencePlanSummary, hasActivePlan }

File: app/plans/get.controller.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    expect(res.render).toHaveBeenCalledWith(`${__dirname}/index`, expected)
+  })
+  it('should set the correct render values when there is no active plan', async () => {
+    req.params.id = 11034
+    getSentencePlanSummary.mockReturnValueOnce(noActivePlan)
+    const expected = { ...displayText, activePlan: false, individualId: 11034 }
+    await controller.sentencePlanSummary(req, res)
+    expect(res.render).toHaveBeenCalledWith(`${__dirname}/index`, expected)
+  })
+  it('should return true when a plan has null or no completed date', () => {
```
Comment (jonhindle) [inline app/plans/get.controller.test.js]: You're testing a different export here; bob it in its own describe.
`descrive('hasActivePlan', () => {`

File: app/plans/displayText.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  heading: 'Active plan',
+  start_plan: 'Start a new plan',
+  no_plan: 'There is no active plan',
+}
```
Comment (jonhindle) [inline app/plans/displayText.js]: What do we gain from this? They're referenced in the nunjucks file with the rest of the text, so the page text ends up split over two files?

File: app/router.js [hunk 6]
```diff
@@ -23,6 +26,7 @@ module.exports = app => {
     })
   })
   app.use(offenderRoot, getOffenderDetails)
-  app.get(offenderRoot, (req, res) => res.render('app/index/index'))
+  app.get(offenderRoot, (req, res) => res.redirect(`/individual-id/${req.params.id}/plans`))
+  app.get(`${offenderRoot}/plans`, (req, res) => sentencePlanSummary(req, res))
```
Comment (jonhindle) [inline app/router.js]: Don't think you need to wrap the function:
app.get(`${offenderRoot}/plans`, sentencePlanSummary)

File: app/router.js [hunk 7]
```diff
@@ -23,6 +26,7 @@ module.exports = app => {
     })
   })
   app.use(offenderRoot, getOffenderDetails)
-  app.get(offenderRoot, (req, res) => res.render('app/index/index'))
+  app.get(offenderRoot, (req, res) => res.redirect(`/individual-id/${req.params.id}/plans`))
```
Comment (jonhindle) [inline app/router.js]: Do we need to redirect? could we not just add the route in below?
app.get([offenderRoot , `${offenderRoot}/plans`], sentencePlanSummary)

File: app/plans/get.controller.js (hunk 1 above)
Comment (Coupey) [reply app/plans/get.controller.js]: Agreed

File: app/plans/get.controller.js (hunk 2 above)
Comment (Coupey) [reply app/plans/get.controller.js]: agreed

File: app/plans/get.controller.test.js (hunk 3 above)
Comment (Coupey) [reply app/plans/get.controller.test.js]: agreed

File: app/plans/get.controller.test.js (hunk 4 above)
Comment (Coupey) [reply app/plans/get.controller.test.js]: agreed

File: app/router.js (hunk 7 above)
Comment (Coupey) [reply app/router.js]: agreed

File: app/router.js (hunk 6 above)
Comment (Coupey) [reply app/router.js]: agreed

File: app/plans/displayText.js (hunk 5 above)
Comment (Coupey) [reply app/plans/displayText.js]: Integrated text into njk file

**Pull request merged**